.github/workflows/rust.yml

```diff
@@ -10,13 +10,34 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+
   build:
+    runs-on: ${{ matrix.os }}
 
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
     - uses: actions/checkout@v3
+
     - name: Build
       run: cargo build --verbose
+
     - name: Run tests
       run: cargo test --verbose
+
+    - name: Rename binary
+      shell: bash
+      run: |
+        if [ "${{ matrix.os }}" = "windows-latest" ]; then
+          mv ./target/debug/e4e-data-management-rust.exe ./target/debug/e4e-data-management-rust-${{ matrix.os }}.exe
+        else
+          mv ./target/debug/e4e-data-management-rust ./target/debug/e4e-data-management-rust-${{ matrix.os }}
+        fi
```

Comment:
It would be ideal if each binary could be traced back to this build - i.e. provide a build time or build date or version in the filename.

Reply:
Currently using commit hash, will make a separate PR for versioning.

Reply:
The commit hash and hash at the end of the filename do not match!

Reply:
Discussed offline but will add here for the record - the Github action performs a merge commit with main and uses that hash, which means the hash used does not exist on the commit tree.